react/features/recording/googleApi/googleElectronClient.js

```diff
@@ -0,0 +1,332 @@
+// @flow
+
+import GoogleProfile from './GoogleProfile';
+
+/**
+ * The key to use for storing the last known access token in session storage.
+ * Session storage is used to at least mitigate having to sign in to Google
+ * often, by eliminating re-sign in on a reload.
+ *
+ * @private
+ * @type {string}
+ */
+const sessionStorageKey = 'googleApiElectronToken';
+
+/**
+ * The account permissions to request access to from Google.
+ *
+ * @private
+ * @type {string}
+ */
+const scopes = [
+    'https://www.googleapis.com/auth/youtube.readonly',
+    'profile',
+    'email'
+].join(' ');
+
+declare var JitsiMeetElectron: Object;
+
+/**
+ * The client ID to be used with the API library.
+ */
+let clientId;
+
+/**
+ * A singleton for loading and interacting with the Google API.
+ */
+const googleApi = {
+    /**
+     * Obtains the global variable used for interacting with the Google API.
+     * This method exists to keep parity with the public interface of the web
+     * implementation of the Google API client.
+     *
+     * @returns {Promise<Object>}
+     */
+    get(): Promise<Object> {
+        return Promise.resolve(this._getGoogleApiClient());
+    },
+
+    /**
+     * Gets the profile for the user associated with the {@link accessToken}.
+     *
+     * @returns {Promise}
+     */
+    getCurrentUserProfile() {
```

Comment:
could stuff like this be put in a common base? (with `isSignedIn` and `_getUserProfile` being overridden by the subclasses?)